src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -32,19 +32,22 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   
-  SmallCityService smallCityService;
+  SmallCityService smallCityService = new SmallCityService();
   private final static Logger LOGGER = Logger.getLogger(DataServlet.class.getName());
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
     String latString = request.getParameter("lat");
     String lngString = request.getParameter("lng");
     double lat = convertToDouble(latString);
     double lng = convertToDouble(lngString);
-    // Google Pittsburgh Office Location (hardcoded prototype)
-    // Will be populated with lat and lng doubles
-    MapLocation userLocation = new MapLocation(40.457410, -79.916573);
-    smallCityService = new SmallCityService(userLocation);
+    MapLocation userLocation = new MapLocation(lat, lng);
+    smallCityService.createUserWithGeolocation(userLocation);
+    } catch(NullPointerException e) {
```

Comment:
Where is the NPE getting thrown in this block? Is it request.getParameter()?

Reply:
Yes

Reply:
Gotcha. What happens if geolocation isn't enabled in the browser, and the zipcode isn't entered? Is this possible? Do you prevent proceeding for users who haven't entered any of these? 

Also, it's a good idea to log something in your catch block here.

Reply:
Right now, if the user presses submit without entering a zipcode, nothing happens (no results). I think this will be more clearer from a UI standpoint once the popup is added when the user enters the site.

Reply:
Ok. Adding a safeguard in the frontend isn't a bad idea.

Please add a log below outlining why the NPE here is ok.

Reply:
I added a logger message and created an issue to track the frontend safeguard #24